hub/util/tests/test_read.py

```diff
@@ -0,0 +1,31 @@
+import io
+import pytest
+import numpy as np
+from PIL import Image  # type: ignore
+from hub.util.load import read
+from hub.util.dataset import get_compressor
+import hub
+
+EXTENSIONS = ("PNG", "JPEG")
+CHECK_META = (True, False)
+parametrize_extension = pytest.mark.parametrize("extension", EXTENSIONS)
+parametrize_check_meta = pytest.mark.parametrize("check_meta", CHECK_META)
+
+
+@parametrize_extension
+@parametrize_check_meta
+def test_read(extension: str, check_meta: bool):
+    img_arr = np.ones((100, 100, 3)).astype("uint8")
+    img = Image.fromarray(img_arr)
+    img.save(f"/tmp/test_read.{extension.lower()}", format=extension)
+    ds = hub.Dataset("/tmp/test_read")
+    ds.create_tensor("image", dtype="uint8")
+    image_dict = read(f"/tmp/test_read.{extension.lower()}", check_meta=check_meta)
+    assert image_dict["shape"] == img_arr.shape
+    assert image_dict["dtype"] == img_arr.dtype
+    assert image_dict["compression"] == extension
```

Comment:
```suggestion
@parametrize_extension
@parametrize_check_meta
@parametrize_all_dataset_storages_and_caches
def test_read(ds: Dataset, extension: str, check_meta: bool):
    img_arr = np.ones((100, 100, 3)).astype("uint8")
    img = Image.fromarray(img_arr)
    img.save(f"/tmp/test_read.{extension.lower()}", format=extension)
    ds.create_tensor("image", dtype="uint8")
    image_dict = read(f"/tmp/test_read.{extension.lower()}", check_meta=check_meta)
    assert image_dict["shape"] == img_arr.shape
    assert image_dict["dtype"] == img_arr.dtype
    assert image_dict["compression"] == extension
```

Reply:
this test should actually be moved into `hub.api.tests.test_api`

and `hub.util.tests.test_read` should not use `Dataset` at all